src/javascript/javascript-export-scanner.ts

```diff
@@ -48,24 +48,50 @@ export class Export implements Resolvable, Feature {
 
   constructor(
       astNode: ExportNode, statementAst: babel.Statement,
-      sourceRange: SourceRange|undefined, nodePath: NodePath<babel.Node>) {
+      sourceRange: SourceRange|undefined, nodePath: NodePath<babel.Node>,
+      exportingAllFrom?: Iterable<Export>) {
     this.astNode = astNode;
     this.statementAst = statementAst;
-    for (const name of esutil.getBindingNamesFromDeclaration(astNode)) {
+    let exportedIdentifiers;
+    if (exportingAllFrom !== undefined) {
+      exportedIdentifiers = (function*() {
+        for (const export_ of exportingAllFrom) {
+          yield* export_.identifiers;
+        }
+      })();
+    } else {
+      exportedIdentifiers = esutil.getBindingNamesFromDeclaration(astNode);
+    }
+    for (const name of exportedIdentifiers) {
       this.identifiers.add(name);
     }
 
     this.astNodePath = nodePath;
     this.sourceRange = sourceRange;
   }
 
-  // It's immutable, and it doesn't care about other documents, so it's
-  // both a ScannedFeature and a Feature. This is just one step in an
-  // arbitrarily long chain of references.
-  resolve(_document: Document): Feature|undefined {
-    // TODO: Could potentially get a speed boost by doing the Reference
-    //   resolution algorithm here. Especially in cases of re-export.
-    //   would need to separate out ScannedExport from Export in that case.
+  // TODO: Could potentially get a speed boost by doing the Reference
+  //   resolution algorithm here, rather than re-doing it every single place
+  //   this export is referenced.
+  resolve(document: Document): Feature|undefined {
+    if (babel.isExportAllDeclaration(this.astNode)) {
+      const [import_] =
+          document.getFeatures({kind: 'import', statement: this.statementAst});
+      if (import_ === undefined) {
```

Comment:
I'd like to make this not be the case... I think Imports should be features whether they resolve or not, but generate a warning and have an undefined URL. Worth taking not of where we make this assumption.

Reply:
I'm cool with that. The type checker will save us here when we make that change.